crates/polars-plan/src/dsl/functions/concat.rs

```diff
@@ -4,15 +4,18 @@ use super::*;
 /// Horizontally concat string columns in linear time
 pub fn concat_str<E: AsRef<[Expr]>>(s: E, separator: &str, ignore_nulls: bool) -> Expr {
     let input = s.as_ref().to_vec();
-    let separator = separator.into();
+    let should_ignore_nulls = if separator.is_empty() { false } else { ignore_nulls };
+    let separator: PlSmallStr = separator.into();
+
+    // For empty separator, we need to ensure nulls make the result null
+    let concat_fn = StringFunction::ConcatHorizontal {
+        delimiter: separator,
+        ignore_nulls: should_ignore_nulls,
```

Comment:
Can just be ignore_nulls && seperator.is_empty()

Reply:
this won't work, because `separator` is will move into `StringFunction::ConcatHorizontal`. and .clone() is too expensive. so a variabled was introuced before moving.

Reply:
It is a reference, not an owned value, and you can just flip the order in the creation of the struct^

Reply:
```rust
separator.is_empty().not() && ignore_nulls;
```
if separator is empty, it will default to false in truth table. else will be whatever ignore_nulls is.